### ISSUE TITLE
Add docs describing how to use SecurityPolicy

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#2072]
*Description*:
>Describe the issue.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by zhaohuabing** [comment]
@arkodg The content of `SecurityPolicy` has been covered in multiple individual tasks:

- Basic Auth: https://gateway.envoyproxy.io/latest/tasks/security/basic-auth/
- CORS: https://gateway.envoyproxy.io/latest/tasks/security/cors/
- Ext Auth: https://gateway.envoyproxy.io/latest/tasks/security/ext-auth/
- JWT Auth: https://gateway.envoyproxy.io/latest/tasks/security/jwt-authentication/
- OIDC: https://gateway.envoyproxy.io/latest/tasks/security/oidc/

Do we want a page to summarize all these individual features within the `SecurityPolicy`?

**Comment by zhaohuabing** [comment]
NM, found answer in this comment https://github.com/envoyproxy/gateway/issues/2070#issuecomment-2116586224

**Comment by arkodg** [comment]
yeah @zhaohuabing this is part of https://github.com/envoyproxy/gateway/issues/2854, linking it as a sub task
